dune_api_scripts/utils.py

```diff
@@ -21,3 +21,32 @@ def store_as_json_file(data_set):
         json.dump(data_set, f, ensure_ascii=False, indent=4)
     print("Written updates into: " + os.path.join(file_path,
           Path("user_data_from" + datetime.now().strftime("%m-%d-%Y") + ".json")))
+
+
+def build_string_for_affiliate_referrals_pairs():
+    file_path = Path(os.environ['APP_DATA_REFERRAL_RELATION_FILE'])
+    app_data_referral_link = json.loads(  # loads one input to create always a valid table.
+        '{"0x0000000000000000000000000000000000000000000000000000000000000abc": "0x0000000000000000000000000000000000000000"}')
+    if file_path.is_file():
+        with open(file_path) as json_file:
+            app_data_referral_link = json.load(json_file)
+
+    # Building value pairs "(appDataHash, referral),"
+    string_of_pair_app_data_referral = ""
+    for hash in app_data_referral_link:
+        string_of_pair_app_data_referral += "('" + hash+"','" + \
+            app_data_referral_link[hash].replace("0", "/", 1)+"'),"
```

Comment:
This is really not a great approach. Here I would propose making use of `",".join()`. Beginning with a dictionary `d`, converting to a list of tuples. Once you have the tuples `[(a, b)]` then you can build individual items with the string formatter  to make items like "(x, y)" as strings and then using `",".join()`. I will try and make an example for us (although I don't know why you are replacing "0" with "/"... OH this is to turn Dune's `/x` into `0x` or vice versa.

Reply:
Example
```py
In [1]: d = { "a": 1, "b": 2, "c": 3 }

In [2]: x = ["({},{})".format(x, y) for (x,y) in d.items()]

In [3]: x
Out[3]: ['(a,1)', '(b,2)', '(c,3)']

In [4]: ",".join(x)
Out[4]: '(a,1),(b,2),(c,3)'
```

So to put all this in one line, you could do:

```py
# Note that I have forgotten to replace the "0" with "/" here (left as an exercise)
return ",".join(["({},{})".format(appHash, referral) for (appHash, referral) in app_data_referral_link.items()])
```

Also, I might change the name `app_data_referral_link` to `referral_map` or something...